dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java

```diff
@@ -967,7 +967,7 @@ private Boolean checkCmdParam(Command command, Map<String, String> cmdParam) {
         if (processInstance.getCommandParam() != null) {
             Map<String, String> processCmdParam = JSONUtils.toMap(processInstance.getCommandParam());
             processCmdParam.forEach((key, value) -> {
-                if (!cmdParam.containsKey(key)) {
+                if (cmdParam != null && !cmdParam.containsKey(key)) {
```

Comment:
You can extend a toMap method with `orDefault` parameter in `JSONUtils` to return an `empty Map,`  replacement with null judgment.

Reply:
Accroding context，  an empty Map may not a good idea

Reply:
Agree with @WangJPLeo , and then we don't need to check null everywhere.
@tracehh Why you think the empty map is not a good idea?

Reply:
I think, if the empty map， and the context code dosen't run